crates/vm/levm/src/memory.rs

```diff
@@ -67,81 +61,39 @@ pub fn try_store_word(memory: &mut Memory, offset: U256, word: U256) -> Result<(
     )
 }
 
-pub fn try_store_data(memory: &mut Memory, offset: U256, data: &[u8]) -> Result<(), VMError> {
-    let new_size = offset
-        .checked_add(data.len().into())
-        .ok_or(OutOfBounds)?
-        .try_into()
-        .map_err(|_err| ExceptionalHalt::VeryLargeNumber)?;
-    try_resize(memory, new_size)?;
-    try_store(memory, data, offset, data.len())
-}
+pub fn try_store_data(memory: &mut Memory, offset: usize, data: &[u8]) -> Result<(), VMError> {
+    let new_size = offset.checked_add(data.len()).ok_or(OutOfBounds)?;
 
-pub fn try_store_range(
-    memory: &mut Memory,
-    offset: U256,
-    size: usize,
-    data: &[u8],
-) -> Result<(), VMError> {
-    if size == 0 {
-        return Ok(());
-    }
-
-    let new_size = offset
-        .checked_add(size.into())
-        .ok_or(OutOfBounds)?
-        .try_into()
-        .map_err(|_err| ExceptionalHalt::VeryLargeNumber)?;
     try_resize(memory, new_size)?;
-    try_store(memory, data, offset, size)
+    try_store(memory, data, offset, data.len())
 }
 
 fn try_store(
     memory: &mut Memory,
     data: &[u8],
-    at_offset: U256,
+    at_offset: usize,
     data_size: usize,
 ) -> Result<(), VMError> {
     if data_size == 0 {
         return Ok(());
     }
 
-    let at_offset: usize = at_offset
-        .try_into()
-        .map_err(|_err| ExceptionalHalt::VeryLargeNumber)?;
+    #[expect(clippy::arithmetic_side_effects, clippy::indexing_slicing)]
+    memory[at_offset..at_offset + data_size].copy_from_slice(data);
```

Comment:
If we're certain this is in bounds, it would be a good idea to add a `reason = "..."` clause to the `expect` explaining why we're certain.

Reply:
Also, we should either also slice `data` or get `data_size` from its `.len()` rather than assuming it matches. Otherwise we may panic due to the slices not having the same size.